libraries/botbuilder-azure-blobs/src/blobsTranscriptStore.ts

```diff
@@ -0,0 +1,230 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import getStream from 'get-stream';
+import pmap from 'p-map';
+import { Activity, PagedResult, TranscriptInfo, TranscriptStore } from 'botbuilder';
+import { maybeCast } from './maybeCast';
+import { sanitizeBlobKey } from './sanitizeBlobKey';
+
+import {
+    ContainerClient,
+    ContainerListBlobHierarchySegmentResponse,
+    StoragePipelineOptions,
+} from '@azure/storage-blob';
+
+// Formats a timestamp in a way that is consistent with the C# SDK
+function formatTicks(timestamp: Date): string {
+    const epochTicks = 621355968000000000; // the number of .net ticks at the unix epoch
+    const ticksPerMillisecond = 10000; // there are 10000 .net ticks per millisecond
+    const ticks = epochTicks + timestamp.getTime() * ticksPerMillisecond;
+    return ticks.toString(16);
+}
+
+function getChannelPrefix(channelId: string): string {
+    return sanitizeBlobKey(`${channelId}/`);
+}
+
+function getConversationPrefix(channelId: string, conversationId: string): string {
+    return sanitizeBlobKey(`${channelId}/${conversationId}`);
+}
+
+// Transforms an activity into its blob name
+function getBlobName(activity: Activity): string {
+    return sanitizeBlobKey(
+        [activity.channelId, activity.conversation.id, `${formatTicks(activity.timestamp)}-${activity.id}.json`].join(
+            '/'
+        )
+    );
+}
+
+/**
+ * @remarks
+ * Each activity is stored as JSON blob with a structure of
+ * `container/{channelId]/{conversationId}/{Timestamp.ticks}-{activity.id}.json`.
+ */
+export class BlobsTranscriptStore implements TranscriptStore {
+    private readonly _containerClient: ContainerClient;
+    private readonly _concurrency: number;
+    private _maxPageSize = 20;
+    private _initializePromise: Promise<unknown>;
+
+    /**
+     * Constructs a BlobsStorage instance.
+     *
+     * @param connectionString Azure Blob Storage connection string
+     * @param containerName Azure Blob Storage container name
+     * @param options Azure Blob Storage [StoragePipelineOptions](xref:@azure/storage-blob.StoragePipelineOptions) options
+     */
+    constructor(connectionString: string, containerName: string, options?: StoragePipelineOptions) {
```

Comment:
I have the same feedback as I relayed in my comment on BlobsStorage's constructor.

Reply:
I'll take the same approach here.